tests/Areas/AppConfig/UnitTests/KeyValue/KeyValueListCommandTests.cs

```diff
@@ -0,0 +1,159 @@
+            .AddSingleton(_appConfigService)
+            .BuildServiceProvider();
+        _context = new(_serviceProvider);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ReturnsSettings_WhenSettingsExist()
+    {
+        // Arrange
+        var expectedSettings = new List<KeyValueSetting>
+        {
+            new() { Key = "key1", Value = "value1", Label = "prod" },
+            new() { Key = "key2", Value = "value2", Label = "dev" }
+        };
+        _appConfigService.ListKeyValues(
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<RetryPolicyOptions>())
+          .Returns(expectedSettings);
+
+        var args = _parser.Parse(["--subscription", "sub123", "--account-name", "account1"]);
+
+        // Act
+        var response = await _command.ExecuteAsync(_context, args);
+
+        // Assert
+        Assert.Equal(200, response.Status);
+        Assert.NotNull(response.Results);
+
+        var json = JsonSerializer.Serialize(response.Results);
+        var result = JsonSerializer.Deserialize<KeyValueListResult>(json, new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+        });
+
+        Assert.NotNull(result);
+        Assert.Equal(2, result.Settings.Count);
+        Assert.Equal("key1", result.Settings[0].Key);
+        Assert.Equal("key2", result.Settings[1].Key);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ReturnsFilteredSettings_WhenKeyFilterProvided()
+    {
+        // Arrange
+        var expectedSettings = new List<KeyValueSetting>
+        {
+            new() { Key = "key1", Value = "value1", Label = "prod" }
+        };
+        _appConfigService.ListKeyValues(
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<string>(),
+          Arg.Any<RetryPolicyOptions>())
+          .Returns(expectedSettings);
+
+        var args = _parser.Parse(["--subscription", "sub123", "--account-name", "account1", "--key", "key1"]);
+
+        // Act
+        var response = await _command.ExecuteAsync(_context, args);
+
+        // Assert
+        Assert.Equal(200, response.Status);
+        await _appConfigService.Received(1).ListKeyValues("account1", "sub123", "key1", null, null, Arg.Any<RetryPolicyOptions>());
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ReturnsFilteredSettings_WhenLabelFilterProvided()
+    {
+        var args = _parser.Parse(["--subscription", "sub123", "--account-name", "account1", "--label", "prod"]);
+
+        // Act
+        var response = await _command.ExecuteAsync(_context, args);
+
+        // Assert
+        Assert.Equal(200, response.Status);
+        await _appConfigService.Received(1).ListKeyValues("account1", "sub123", null, "prod", null, Arg.Any<RetryPolicyOptions>());
+    }
```

Comment:
We should have the test return an object and then verify its contents like suggested for the one above.